internal/controllers/certmanager.go

```diff
@@ -223,6 +232,62 @@ func (r *Reconciler) createOrUpdateIssuer(ctx context.Context, issuer *certv1.Is
 	return nil
 }
 
+func (r *Reconciler) issuerCAChanged(current *certv1.CAIssuer, updated *certv1.CAIssuer) bool {
+	// Compare the .spec.ca.secretName in the current and updated Issuer. Return whether they differ.
+	if current == nil {
+		return false
+	}
+	currentSecret := current.SecretName
+	updatedSecret := ""
+	if updated != nil {
+		updatedSecret = updated.SecretName
+	}
+
+	if currentSecret != updatedSecret {
+		r.Log.Info("certificate authority has changed, deleting issued certificates",
+			"current", currentSecret, "updated", updatedSecret)
+		return true
+	}
+	return false
+}
+
+func (r *Reconciler) deleteCertChain(ctx context.Context, namespace string, caSecretName string, owner metav1.Object) error {
+	// Look up all certificates in this namespace
+	certs := &certv1.CertificateList{}
+	err := r.Client.List(ctx, certs, &client.ListOptions{
+		Namespace: namespace,
+	})
+	if err != nil {
+		return err
+	}
+
+	for i, cert := range certs.Items {
+		// Is the certificate owned by this CR, and not the CA itself?
+		if metav1.IsControlledBy(&certs.Items[i], owner) && cert.Spec.SecretName != caSecretName {
+			// Clean up secret referenced by the cert
+			secret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      cert.Spec.SecretName,
+					Namespace: cert.Namespace,
+				},
+			}
+
+			err := r.deleteSecret(ctx, secret)
+			if err != nil {
+				return err
+			}
+			// Delete the certificate
+			err = r.Client.Delete(ctx, &certs.Items[i])
+			if err != nil {
+				return err
+			}
```

Comment:
```suggestion
			if err != nil  && !kerrors.IsNotFound(err) {
				return err
			}
```

Not likely but safe to check?

Reply:
Sure, that shouldn't hurt. I've done this now.